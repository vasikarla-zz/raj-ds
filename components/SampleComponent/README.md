# @raj-ds/sample-component

Welcome to my `SampleComponent` component.

## Installation

```sh
npm i @raj-ds/sample-component
# or with yarn
yarn add @raj-ds/sample-component
```

Then to use the component in your code just import it!

```js
import SampleComponent from '@raj-ds/sample-component';
// and with css-modules
import '@raj-ds/sample-component/dist/main.css';
```
