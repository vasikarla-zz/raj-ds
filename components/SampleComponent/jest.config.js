const base = require('@design-systems/test/jest.config.base');
const { name } = require('./package.json');

module.exports = {
  ...base,
  name,
  displayName: name
};
