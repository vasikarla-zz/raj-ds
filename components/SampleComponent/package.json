{
  "name": "@raj-ds/sample-component",
  "version": "0.0.0",
  "main": "./dist/cjs/index.js",
  "module": "./dist/esm/index.js",
  "types": "./dist",
  "repository": "git@github.com:vasikarla/raj-ds.git",
  "author": "Raj Vasikarla raj_vasikarla@intuit.com",
  "contributors": ["Raj Vasikarla raj_vasikarla@intuit.com"],
  "license": "MIT",
  "sideEffects": ["*.css"],
  "scripts": {
    "clean": "ds clean",
    "build": "ds build",
    "dev": "ds dev",
    "storybook": "ds storybook start",
    "start": "ds build --watch",
    "test": "ds test",
    "lint": "ds lint",
    "size": "ds size"
  },
  "dependencies": {
    "@babel/runtime": "7.7.2"
  },
  "peerDependencies": {
    "react": ">= 16.8.6"
  },
  "files": [
    "dist",
    "src",
    "!*.snippet.*",
    "!*theme.*",
    "!*.test.*",
    "!__snapshots__",
    "!__tests__"
  ]
}
