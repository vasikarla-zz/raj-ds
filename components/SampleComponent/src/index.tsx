import React from 'react';
import styles from './SampleComponent.css';

interface SampleComponentProps {
  /** The value for the component to display */
  value?: string;
}

/** A simple component. */
export const SampleComponent = (props: SampleComponentProps) => {
  return <div className={styles.sampleComponent}>This is a SampleComponent component {props.value}</div>;
}

export default SampleComponent;
