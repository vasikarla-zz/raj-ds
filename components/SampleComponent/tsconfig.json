{
  "extends": "../../tsconfig.build.json",
  "include": ["src/**/*", "../../typings/**/*"],

  "compilerOptions": {
    "outDir": "./dist",
    "rootDir": "./src",
    "composite": true,

    // Need to override emotion's types because they pollute the global
    "baseUrl": ".",
    "paths": {
      "@emotion/core": ["../../typings/emotion.d.ts"]
    }
  }
}
