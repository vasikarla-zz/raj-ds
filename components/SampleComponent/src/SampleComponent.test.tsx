import React from 'react';
import { render } from '@testing-library/react';
import SampleComponent from '.';

describe('SampleComponent', () => {
  test('It matches the snapshot', () => {
    const { container } = render(<SampleComponent />);
    expect(container).toMatchSnapshot();
  });  
});