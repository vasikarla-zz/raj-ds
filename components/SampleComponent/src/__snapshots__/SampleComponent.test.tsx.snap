// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SampleComponent It matches the snapshot 1`] = `
<div>
  <div
    class="sampleComponent"
  >
    This is a SampleComponent component 
  </div>
</div>
`;
